face/src/main/java/io/blockv/face/client/FaceManager.kt

```diff
@@ -0,0 +1,83 @@
+package io.blockv.face.client
+
+import io.blockv.core.model.Action
+import io.blockv.core.model.Face
+import io.blockv.core.model.Vatom
+
+interface FaceManager {
+
+  enum class EmbeddedProcedure(val viewMode: String, val fallback: EmbeddedProcedure?) {
+
+    ICON("icon", null),
+    ACTIVATED("activated", ICON),
+    FULLSCREEN("fullscreen", ICON),
+    CARD("card", ICON);
+
+    val routine: FaceSelectionProcedure
```

Comment:
You still have 'routine' language - should it be 'procedure' rather? On iOS, there is no mention of routine.

Reply:
"routine" is for internal use but is still public visible. Don't like the sound of EmbeddedProcedure.ICON.procedure but can change it to that if you prefer.

Reply:
As you say it's internal so does not matter too much. EmbeddedProcedure is an enum of case that map to procedures, so `EmbeddedProcedure.ICON.procedure` is better imo – but I agree does look odd.